src/Core/src/Platform/iOS/DatePickerExtensions.cs

```diff
@@ -25,6 +25,8 @@ public static void UpdateDate(this MauiDatePicker nativeDatePicker, IDatePicker
 				picker.SetDate(datePicker.Date.ToNSDate(), false);
 
 			nativeDatePicker.Text = datePicker.Date.ToString(datePicker.Format);
+
+			nativeDatePicker.UpdateCharacterSpacing(datePicker);
```

Comment:
Need to apply the formatting. Both this and TimePicker need to actually refactor this call into a UpdateFormatting or something because we will forget to add things here when we add more properties.